libs/MeshKernelApi/tests/src/CurvilinearGridTests.cpp

```diff
@@ -1433,3 +1433,40 @@ TEST(CurvilinearGrid, SnapToSpline)
         EXPECT_NEAR(curvilinearGrid.node_y[i], originalNodeY[i], tolerance);
     }
 }
+
+TEST(CurvilinearGrid, CurvilinearGetBoundariesAsPolygons_ShouldGetBoundariesAsPolygons)
+{
+    // Prepare
+    int meshKernelId;
+    auto errorCode = meshkernelapi::mkernel_allocate_state(0, meshKernelId);
+    ASSERT_EQ(meshkernel::ExitCode::Success, errorCode);
+
+    MakeGridParameters makeGridParameters;
+
+    makeGridParameters.num_columns = 10;
+    makeGridParameters.num_rows = 10;
+    makeGridParameters.angle = 0.0;
+    makeGridParameters.origin_x = 0.0;
+    makeGridParameters.origin_y = 0.0;
+    makeGridParameters.block_size_x = 10.0;
+    makeGridParameters.block_size_y = 10.0;
+
+    errorCode = meshkernelapi::mkernel_curvilinear_compute_rectangular_grid(meshKernelId, makeGridParameters);
+    ASSERT_EQ(meshkernel::ExitCode::Success, errorCode);
+
+    int numberOfPolygonNodes;
+    errorCode = meshkernelapi::mkernel_curvilinear_count_boundaries_as_polygons(meshKernelId, 0, 0, 9, 9, numberOfPolygonNodes);
+    ASSERT_EQ(meshkernel::ExitCode::Success, errorCode);
+
+    meshkernelapi::GeometryList boundaryPolygon;
+    std::vector<double> coordinates_x(numberOfPolygonNodes);
+    std::vector<double> coordinates_y(numberOfPolygonNodes);
+
+    boundaryPolygon.coordinates_x = coordinates_x.data();
+    boundaryPolygon.coordinates_y = coordinates_y.data();
+    boundaryPolygon.num_coordinates = numberOfPolygonNodes;
+    boundaryPolygon.geometry_separator = constants::missing::doubleValue;
+
+    errorCode = mkernel_curvilinear_get_boundaries_as_polygons(meshKernelId, 0, 0, 9, 9, boundaryPolygon);
+    ASSERT_EQ(meshkernel::ExitCode::Success, errorCode);
```

Comment:
add some more assertions, e.g. of the sizes of all vectors in boundaryPolygon.

also add a test with lower-right and upper-left for example.